.github/workflows/ci.yml

```diff
@@ -0,0 +1,68 @@
+name: ci
+
+on:
+  pull_request:
+    paths-ignore:
+      - "**/*.asciidoc"
+      - "**/*.md"
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - "**/*.asciidoc"
+      - "**/*.md"
+
+jobs:
+  sanity-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: elastic/apm-pipeline-library/.github/actions/pre-commit@current
+
+  create-test-matrix:
+    outputs:
+      default-matrix: ${{ steps.generate.outputs.matrix }}
+      main-matrix: ${{ steps.generate-main.outputs.matrix }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - id: generate
+        uses: elastic/apm-pipeline-library/.github/actions/version-framework@current
+        with:
+          versionsFile: .ci/.jenkins_ruby.yml
+          frameworksFile: .ci/.jenkins_framework.yml
+          excludedFile: .ci/.jenkins_exclude.yml
+      - id: generate-main
+        uses: elastic/apm-pipeline-library/.github/actions/version-framework@current
+        with:
+          versionsFile: .ci/.jenkins_ruby.yml
+          frameworksFile: .ci/.jenkins_main_framework.yml
+          excludedFile: .ci/.jenkins_exclude.yml
+
+  default:
+    needs:
+      - create-test-matrix
+    uses: ./.github/workflows/run-matrix.yml
+    with:
+      matrix: ${{ toJSON(fromJSON(needs.create-test-matrix.outputs.default-matrix)) }}
+
+  main:
```

Comment:
ditto

Reply:
https://github.com/elastic/apm-agent-ruby/pull/1366/commits/860c5eb476658766deb5186cea4a1e8134e26a57

I added a comment. But I also refactored the "sub workflow" to make this a little bit clearer.